### PR TITLE
Bug 1237496 - Use mozIntl.UnitFormat in System

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1237496

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f0f49526d8becf0123c5b756f6bb8421c8f99112

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5ccf684350e693e52ca6d86e555a2b6298877ca8

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ca7339f2fb4cb35f9a5f25cd067d1144a375f1fb

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=39b0928a97d9b31a300d024a51cdfe909ded25eb

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8ff714772771e6a786eba8e473f5e4c79b4ad828

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2d8a38ceddaa6b8da13b5a305cf7b5d1c8b406b5

File: apps/system/js/app_install_manager.js [hunk 1]
```diff
@@ -242,7 +243,9 @@
 
       if (manifest.size) {
         this.size.removeAttribute('data-l10n-id');
-        this.size.textContent = this.humanizeSize(manifest.size);
+        this.humanizeSize(manifest.size).then(val => {
+          this.size.textContent = val;
+        });
```
Comment (stasm) [inline apps/system/js/app_install_manager.js]: Wouldn't it be a better approach to pass this through l20n.js?  In the future we'd like to get rid of `humanizeSize` altogether and replace it with in-localization macros/filters.

Comment (stasm) [inline apps/system/js/app_install_manager.js]: Also, a style nit: you don't need curly braces in the arrow function if it only has one expression.

File: apps/system/js/device_storage_watcher.js [hunk 2]
```diff
@@ -157,26 +143,22 @@
...
-          req.onsuccess = () => {
-            if (typeof req.result !== 'undefined') {
-              this._freeSpace = this.formatSize(req.result);
-            }
+          return this._appStorage.freeSpace().then(result => {
+            this._freeSpace = result;
             if (this._lowDeviceStorage) {
-              this.updateAvailableSpace(this._freeSpace);
-              return;
+              console.log('calling updateAvailableSpace');
```
Comment (stasm) [inline apps/system/js/device_storage_watcher.js]: Remove?

File: apps/system/locales/system.en-US.properties [hunk 3]
```diff
@@ -532,16 +526,15 @@ notification-powersaving-mode-on-description=Power saving mode is on.
...
-# in the corresponding unit ({{unit}}), which is one of the byteUnit-* strings
-# of this same file.
-free-space={[ plural(value) ]}
-free-space[zero]=No remaining free space.
-free-space[one]=Less than {{value}} {{unit}} remaining.
-free-space[two]=Less than {{value}} {{unit}} remaining.
-free-space[few]=Less than {{value}} {{unit}} remaining.
-free-space[many]=Less than {{value}} {{unit}} remaining.
-free-space[other]=Less than {{value}} {{unit}} remaining.
+# LOCALIZATION NOTE {{value}} is the amount of remaining free space
```
Comment (stasm) [inline apps/system/locales/system.en-US.properties]: Mention that `value` is a string and already contains the unit?

File: apps/system/js/app_install_manager.js (hunk 1 above)
Comment (zbraniecki) [inline apps/system/js/app_install_manager.js]: > Wouldn't it be a better approach to pass this through l20n.js?

I'm not sure if I understand. How can I format unit via l20n.js today?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1f780f8cdd95afb114a5ff166ee8ff49d0fc8c56

**Pull request merged**